internal/lib/utils.go

```diff
@@ -95,38 +101,53 @@ func GetSvcKeysForNodeCRUD() (svcl4Keys []string, svcl7Keys []string) {
 
 }
 
-func GetPodsFromService(namespace, serviceName string) []utils.NamespaceName {
+func GetPodsFromService(namespace, serviceName string, targetPortName intstr.IntOrString) ([]utils.NamespaceName, int32) {
 	var pods []utils.NamespaceName
+	var targetPort int32
 	svcKey := namespace + "/" + serviceName
 	svc, err := utils.GetInformers().ServiceInformer.Lister().Services(namespace).Get(serviceName)
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
-			return pods
+			return pods, targetPort
 		}
 		if found, podsIntf := objects.SharedSvcToPodLister().Get(svcKey); found {
-			savedPods, ok := podsIntf.([]utils.NamespaceName)
+			savedPods, ok := podsIntf.(PodsWithTargetPort)
 			if ok {
-				return savedPods
+				return savedPods.Pods, savedPods.TargetPort
 			}
 		}
-		return pods
+		return pods, targetPort
 	}
 
 	if len(svc.Spec.Selector) == 0 {
-		return pods
+		return pods, targetPort
 	}
 
 	podList, err := utils.GetInformers().PodInformer.Lister().Pods(namespace).List(labels.SelectorFromSet(labels.Set(svc.Spec.Selector)))
 	if err != nil {
 		utils.AviLog.Warnf("Got error while listing Pods with selector %v: %v", svc.Spec.Selector, err)
-		return pods
+		return pods, targetPort
+	}
+	targetPortFound := false
+	if targetPortName.Type == intstr.Int {
+		targetPortFound = true
+		targetPort = int32(targetPortName.IntValue())
 	}
 	for _, pod := range podList {
+		if !targetPortFound {
+			for _, pc := range pod.Spec.Containers {
```

Comment:
I think matching with container Port may not be necessary. Kubernetes doesn't mandate the Pod to have container port specified.
 @chauhanshubham we got a comment regarding this in Antrea and had to remove that check.